packages/engine-formula/src/functions/math/asin/index.ts

```diff
@@ -0,0 +1,68 @@
+/**
+ * Copyright 2023-present DreamNum Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { ErrorType } from '../../../basics/error-type';
+import type { BaseValueObject } from '../../../engine/value-object/base-value-object';
+import { ErrorValueObject } from '../../../engine/value-object/base-value-object';
+import { NumberValueObject } from '../../../engine/value-object/primitive-object';
+import { BaseFunction } from '../../base-function';
+
+export class Asin extends BaseFunction {
+    override minParams = 1;
+
+    override maxParams = 1;
+
+    override calculate(variant: BaseValueObject) {
+        if (variant.isString()) {
+            variant = variant.convertToNumberObjectValue();
+        }
+
+        if (variant.isError()) {
+            return variant;
+        }
+
+        if ((variant as BaseValueObject).isArray()) {
+            return (variant as BaseValueObject).map((currentValue) => {
+                if (currentValue.isError()) {
+                    return currentValue;
+                }
+                return asin(currentValue as BaseValueObject);
+            });
+        }
+
+        return asin(variant as BaseValueObject);
+    }
+}
+
+function asin(num: BaseValueObject) {
+    let currentValue = num.getValue();
+
+    if (num.isBoolean()) {
+        currentValue = currentValue ? 1 : 0;
+    }
+
+    if (!Number.isFinite(currentValue) || Number(currentValue) < -1 || Number(currentValue) > 1) {
+        return ErrorValueObject.create(ErrorType.VALUE);

```

Comment:
should reports #NUM!
![image](https://github.com/dream-num/univer/assets/26371161/dd4d70ec-80dc-4eac-96ab-4f8ca455443e)